src/DotVVM.Framework/Diagnostics/DotvvmDiagnosticsConfiguration.cs

```diff
@@ -9,24 +9,41 @@ namespace DotVVM.Framework.Diagnostics
 
     public class DotvvmDiagnosticsConfiguration
     {
-        public DotvvmDiagnosticsConfiguration()
+       
+
+        private DiagnosticsServerConfiguration configuration;
+        private DateTime configurationLastWriteTimeUtc;
+
+        public string DiagnosticsServerHostname
         {
-            LoadConfiguration();
+            get
+            {
+                EnsureConfigurationValid();
+                return configuration?.HostName;
+            }
         }
 
-        private DiagnosticsServerConfiguration configuration;
 
-        public string DiagnosticsServerHostname => configuration?.HostName;
 
-        public int? DiagnosticsServerPort => configuration?.Port;
+        public int? DiagnosticsServerPort
+        {
+            get
+            {
+                EnsureConfigurationValid();
+                return configuration?.Port;
+            }
+        }
 
-        private void LoadConfiguration()
+        private void EnsureConfigurationValid()
```

Comment:
I think that this should be called `LoadConfiguration` as it really loads it instead of just validating it.

Reply:
What about `EnsureConfigurationLoaded`?

Reply:
Ok, it's at least not misleading.

Reply:
No, even `EnsureConfigurationLoaded` is totally misleading - it feel to me that it loads the configuration only once while this method realoads every time. Maybe it should be rather called `ReloadConfiguration`